### PR TITLE
Remove [Play All] / [Shuffle] from AA Playlists root (#245)

### PR DESCRIPTION
**Pull request by kimptoc**
## Summary

Removes the `[Play All]` and `[Shuffle]` action rows from the top of the Playlists browse root in Android Auto, bringing it in line with the Albums / Genres / Artists / Decades roots that already carry no top-level actions. Cross-playlist shuffle destroyed the curation that motivates having playlists in the first place, and was today's gateway into the stale-m3u-URI failure cascade tracked in #244.

Per-playlist `[Play All]` / `[Shuffle]` actions inside an opened playlist are unchanged, as are the same actions on the Songs root (a deliberate exception — flat list, not a category browser).

The now-unreachable handler code that asynchronously flattened all playlists into one queue is removed in the same branch (`resolveTracksForListKey` PLAYLISTS_ID branch, `resolvePlaylistNameForListKey` branch, and the two coroutine imports that were only used there).

## Commits

- `1d4f7bc` Remove `[Play All]` / `[Shuffle]` from Playlists root in AA browse — production change + regression test.
- `5e004f8` Cover zero-discovered-playlists path in playlists-root test — secondary test for the empty-playlists path.
- `203c087` Drop dead PLAYLISTS_ID handler branches after removing root actions — hygiene cleanup.

## Test Plan

- [x] `buildMediaItems_playlistsRoot_hasNoPlayAllOrShuffleEntries` — two discovered playlists, asserts no `action:play_all:*` / `action:shuffle:*` entries at playlists root.
- [x] `buildMediaItems_playlistsRoot_hasNoPlayAllOrShuffleEntries_whenNoDiscoveredPlaylists` — empty playlists list, same assertion.
- [x] `./gradlew :shared:testDebugUnitTest` passes.
- [x] `./gradlew :shared:assembleDebug :mobile:assembleDebug` passes.
- [ ] Manual: AA → Playlists root — only user playlists and smart playlists visible, no `[Play All]` / `[Shuffle]` rows.
- [ ] Manual: AA → a specific playlist — `[Play All]` / `[Shuffle]` still present inside.
- [ ] Manual: AA → Songs root — `[Play All]` / `[Shuffle]` still present.

## Notes

- Test uses the string literal `"playlists"` rather than `MyMusicService.PLAYLISTS_ID` because that constant is `private` on `main`. PR #243 changes its visibility to `internal` — if #243 lands first, this test can be cleaned up on rebase.
- The whole-branch reviewer noted that the secondary test's commit message slightly overpromises ("tighten the regression net"): the deleted guard's second condition (`cachedFiles.isNotEmpty()` with empty playlists) is not directly exercised. Test 1 carries the full regression weight. Not blocking, flagged here for transparency.
- Broader regression coverage for #243's findings (manifest permission, FGS-in-onCreate) is tracked separately as #247.

Fixes #245.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: shared/src/main/java/com/example/mymediaplayer/shared/MyMusicService.kt [hunk 1]
```diff
@@ -1011,38 +1009,14 @@ class MyMusicService : MediaBrowserServiceCompat() {
...
+        return playlistEntriesForBrowse(mediaCacheService.discoveredPlaylists).map { entry ->
             val description = MediaDescriptionCompat.Builder()
                 .setMediaId(entry.mediaId)
                 .setTitle(entry.title)
                 .setIconUri(resourceIconUri(R.drawable.ic_auto_playlists))
                 .build()
             MediaItem(description, MediaItem.FLAG_BROWSABLE)
-        }
-        return items
+        }.toMutableList()
```
[the remark this answers is not in the document]
Comment (kimptoc) [reply shared/src/main/java/com/example/mymediaplayer/shared/MyMusicService.kt:1066]: Acknowledged, and you're right that the `.toMutableList()` produces a copy nothing mutates.

Declining the narrow fix here because all 25+ `buildXItems` functions in this file (`buildSongsItems`, `buildAlbumsItems`, `buildMediaItemsForArtist`, `buildHomeItems`, `buildMediaItems` itself, etc.) currently declare `MutableList<MediaItem>` returns. Changing just `buildPlaylistsItems` to `List<MediaItem>` would make it the lone exception against an otherwise uniform convention — net code-health negative.

The right way to address this is a single file-wide refactor of the whole `buildXItems` chain (including `buildMediaItems`'s signature, which is the called API) — that's out of scope for #245, which is a UI affordance cleanup. Happy to file a separate code-health issue if you want it tracked.

Per-browse cost is one copy of ~10 entries; negligible relative to the inconsistency cost of half-fixing it.

[the remark this answers is not in the document]
Comment (kimptoc) [reply shared/src/main/java/com/example/mymediaplayer/shared/MyMusicService.kt:1066]: Follow-up filed: #249 (file-wide `buildXItems` → `List<MediaItem>` cleanup).
